src/main/msp/msp.c

```diff
@@ -2742,6 +2765,30 @@ static mspResult_e mspProcessInCommand(mspDescriptor_t srcDesc, uint8_t cmdMSP,
         break;
 #endif
 
+#ifdef USE_TUNING_SLIDERS
+    // Added in MSP API 1.43
+    case MSP_SET_TUNING_SLIDERS:
+        {
+            currentPidProfile->slider_pids_mode = sbufReadU8(src);
+            currentPidProfile->slider_master_multiplier = sbufReadU8(src);
+            currentPidProfile->slider_roll_pitch_ratio = sbufReadU8(src);
+            currentPidProfile->slider_i_gain = sbufReadU8(src);
+            currentPidProfile->slider_pd_ratio = sbufReadU8(src);
+            currentPidProfile->slider_pd_gain = sbufReadU8(src);
+            currentPidProfile->slider_dmin_ratio = sbufReadU8(src);
+            currentPidProfile->slider_ff_gain = sbufReadU8(src);
+
+            currentPidProfile->slider_dterm_filter = sbufReadU8(src);
+            currentPidProfile->slider_dterm_filter_multiplier = sbufReadU8(src);
+
+            gyroConfigMutable()->slider_gyro_filter = sbufReadU8(src);
+            gyroConfigMutable()->slider_gyro_filter_multiplier = sbufReadU8(src);
+
+            applyTuningSliders(currentPidProfile);
+        }
+        break;
+#endif
+
```

Comment:
I just had a quick scan of this code but I don't like this fixed order approach in the MSP commands.  Using a fixed list of things in the protocol has been shown to be un-workable and problematic for MSP client authors over-time.  As firmware changes things get added/removed, but the MSP command ideally should not change.
For example, the MSP_BOX related commands allows us to add new named modes without having to change the MSP_BOX related commands itself.  They're not a great example, but the concept of a command that doesn't require any maintenance or changes over time is what we're looking for.

suggest assigning a fixed list of identifiers for each 'slider', and having the set command accept the is and the value.  add a mapping function to go from the ID to the offset in the structure you need to change.  response with suitable error codes if the ID is invalid.

MSP_TUNING_SLIDERS should also be updated to send the ID of the slider and the value.

The ID's used should be IDs that are permenently allocated, or dynamically allocated and available with a name (like the MSP_BOX_NAMES).  If a new slider is added allocate a new ID.  Never change the meaning of a slider.  Slider ID's likely should be defined as UINT16_T.

Reply:
@hydra: Can you please explain a bit more what is broken with this so that it cannot be merged, beyond your opinion?

Reply:
as per my first paragraph above.  it's not opinion, it's fact based on prior history of MSP commands.

Reply:
also, for a 3rd party client developer talk with Bart from EZIOSoft who will tell you tales of nightmare regarding every-changing MSP commands in early versions of Betaflight.

Reply:
@hydra If we had more data associated with each slider I'd agree with you. But in this case there's only a single byte value for each slider so I can't see the justification in all the extra code and effort for using some generic ID-based structure. Also adding this complexity will make Lua support (one of the goals) extremely difficult.

Reply:
@etracer65 It's really not that difficult to implement.   just a 'for' loop, an enum for the ids and a map with an id and an offset_of into the structure...

Reply:
I've been busy for couple of weeks, what are your thought on this? @mikeller 
Should I be looking to implement this as suggested or keep it the way it is?

Reply:
@IvoFPV I'd suggest your proceed as-is and revisit later unless something in your approach will make revisiting impossible.

Reply:
my comments still stand.

Some form of fixed ID or name based system is a better long-term approach, the concept of a 'slider' seems like a very generic thing to me.

it feels like the items being set in this are somewhat arbitrary.  additionally the values used have defined min/max settings which are not exposed by the API.  for instance `slider_pids_mode` is defined in settings.c and uses a table for the values, whereas `slider_i_gain` has no min/max other than the limits of the type the value is stored in.  client authors cannot add support for such items in a generic way.

perhaps another approach is to expose the values in `valueTable` that can be set via MSP.  e.g. add the items to `valueTable` as per normal, use them as per normal and add fixed ID's and a list/get/set to each thing that can be set via MSP?  at least that approach would then be generic and logic-free and not require client authors to update their clients when the values in the proposed `MSP_TUNING_SLIDERS` changes.

Reply:
@IvoFPV: As @etracer65 and @fujin say, the current approach is in line with the approach that was taken in Betaflight ever since it split from cleanflight, and it's been working well enough for parameters with little change over time to their specification.
We'll need to look into solving this issue in a more generic way, as the main issue with current MSP  is that all messages are hand crafted - the way to address this will be to move to a system where parameters, and the MSP containing them, including client stubs, are programmatically generated based on a definition.

Reply:
I'm almost one year later to get here, sorry, but only to give my opinion, I think too that is better to go with a "dynamic" way to send the data. This will make the maintenance in the Configurator easier if we decide to change the sliders at some time (add/remove/change the way some of them work).
If we are sure this will be very stable, is not a problem to be fixed, if not is better to send something like:
- Number of sliders (with a 0 or -1 to disable)
- For each slider
   - Acronim
   - Value

And one command for PIDs and another one for Filters. Or have a parameter in the command to filter for one of them. 
But as I say, this was only to give my opinion and not comment about it once merged. If we continue with fixed elements, to simplify things here, we will adapt in the Configurator as always.

Reply:
@McGiverGim: Not sure I agree that this will make maintenance on the configurator side easier. There are different ways how such a 'dynamic' way can work out that are already in use in MSP:
1. the box / switch way, where each switch has exactly the same configuration, is characterised by a name only, and has got no dependencies. This is working relatively well;
2. the way that PID and related values have been done (and rate values for the different rate types are done in a similar way) - these values have different connotations, different ranges, and also different dependencies. The fact that they were done in a dynamic way is actually more confusing than helpful.

To me it looks like the groups / sliders will be more of a 2. type of things, where individual groups will have more context and dependencies than just their name.

Reply:
Yes, they have a big dependency with the values they change:
- After each move of the slider, we need to "ask" the FC for the new values of the PIDs (or the filters, or any other thing we can have in a future if we continue adding more sliders).

I think all the sliders must be equal in the rest of things (ranges, minimum, maximum, etc.)
So they need to be related with something. This is the reason why I ask to use different MSP commands, or to add a parameter more to the Slider with "type" (now it can be pid or filter only).
In this way the use will be more like:
- The user moves a slider of type "PID", we send the slider value to the FC, we read the new PIDs value, we refresh the page.

The advantage of being dynamic is that we don't need to have a lot of "if semver", or code for each of the values (we will make use of them in an array form), and users can customize in code some of them, or disable them totally and it will work. 
But if we think this sliders will be more or less static then the difference is very little. What I don't want is to go and change them each version, adding or removing some of them.

Sorry if all of this has been discussed earlier, the PR is large and I don't understand all.